### PR TITLE
[CA-592] Throw JSON exceptions instead of HTML ones

### PR DESCRIPTION
**Pull request by marctalbott**
This PR is mainly aimed at re-implementing the default HTML exceptions that Flask was throwing as JSON exceptions so that they would be (nearly) identical to the errors that we threw when we used endpoints. This is part of CA-592 since it was initially left off of the work done for that ticket, but was necessary for tests to pass.

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/bond/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/DataBiosphere/bond/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/DataBiosphere/bond/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary. Documentation PRs only need 1 thumb.
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by dvoet — APPROVED** [review]

File: automation/helpers/json_responses.py [hunk 1]
```diff
@@ -548,7 +548,7 @@
               "message": {
                 "type": "string",
                 "enum": [
-                  "Error parsing ProtoRPC request (Unable to parse request content: Message CombinedContainer is missing required field oauthcode)"
+                  "The browser (or proxy) sent a request that this server could not understand."
```
Comment (andy7i) [inline automation/helpers/json_responses.py:551]: With the new error messages, it seems like we're losing informative error messaging which is too bad. I understand if that's outside the scope of this ticket though

File: json_exception_handler.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+from flask import jsonify
+from werkzeug import exceptions
+
+
+class JsonExceptionHandler(object):
+    def __init__(self, app=None):
+        if app:
+            self.init_app(app)
+
+    def json_dict(self, status_code, message):
```
Comment (andy7i) [inline json_exception_handler.py:10]: What does `json_dict` do if the status code is not one of the codes listed? Should we allow that to pass through?

**Review by andy7i — COMMENTED** [review]
drive by comments

**Review by gpolumbo-broad — APPROVED** [review]

File: automation/helpers/json_responses.py (hunk 1 above)
Comment (marctalbott) [reply automation/helpers/json_responses.py:551]: Yeah, unfortunately trying to get Flask and webargs to re-generate the same error message that endpoints and protorpc were generating was too much of a headache so I left it like this. [CA-500](https://broadworkbench.atlassian.net/browse/CA-500) should address this.

**Pull request merged**